src/main/java/de/hysky/skyblocker/mixins/EntityMixin.java

```diff
@@ -84,4 +84,11 @@ private void onTick(CallbackInfo ci) {
 			}
 		}
 	}
+
+	@Inject(method = "setSwimming", at = @At("HEAD"), cancellable = true)
+	private void preventSwimming(boolean swimming, CallbackInfo ci) {
+		if (Utils.isOnSkyblock() && SkyblockerConfigManager.get().general.preventSwimming) {
```

Comment:
Check is `swimming` is true to only block set true calls.

Reply:
Sorry but I want to mention that I've realized that NoFrils' AntiSwim implementation is generally better, and mine causes visual glitches. But I'm not sure if we'll run into problems if we copy his code. Are we sure we can merge these features (e.g. finally add a new category for legacy hitbox fixes and movement fixes)?

Reply:
If you're copying their code, please get their permission. Also if this implementation has visual glitches, I don't think we'll take it.

Reply:
Yeah I may need to close this because I feel that things can't be better solved at the moment. Based on previous discussions, this type of feature that modifies game behavior may not be natively appropriate for Skyblocker (or just currently), and other mods are good enough.